worlds/AutoWorld.py

```diff
@@ -123,8 +123,8 @@ def __new__(mcs, name: str, bases: Tuple[type, ...], dct: Dict[str, Any]) -> Web
             assert group.options, "A custom defined Option Group must contain at least one Option."
             # catch incorrectly titled versions of the prebuilt groups so they don't create extra groups
             title_name = group.name.title()
-            if title_name in prebuilt_options:
-                group.name = title_name
+            assert title_name not in prebuilt_options or title_name == group.name, \
```

Comment:
i'm fine with the `assert title_name == group.name` since i was on the fence about whether that should exist but the other bool here is wrong and the if immediately after this will now not be hit ever.

Reply:
i thought that was an and, not an or 🤦